### PR TITLE
EL-1292: In CircleCi, run tests in parallel - WIP

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1292)

## What changed and why

add parallelism to circleci for running specs
remove 100% coverage requirement on circle
store the test results
add new job to collate results
add new rake task to collate results and ensure 100% coverage

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: Rakefile
Comment (MazOneTwoOne) [inline Rakefile:1]: question - we now have an empty Rakefile, what does it do?

File: .circleci/config.yml [hunk 1]
```diff
@@ -278,7 +313,6 @@ workflows:
           environment: ccq-uat
           context: laa-check-client-qualifies-uat
           requires:
-            - linters
```
Comment (MazOneTwoOne) [inline .circleci/config.yml:281]: comment - hopefully you agree we should have 2 `linters` in this requires section.

File: .circleci/config.yml [hunk 2]
```diff
@@ -136,7 +136,9 @@ jobs:
           template: basic_fail_1
           branch_pattern: main
   run_specs:
+    parallelism: 4
     executor: test-executor
+    resource_class: small
```
Comment (MazOneTwoOne) [inline .circleci/config.yml:141]: comment - thinking about it, we possible don't event need to put `resource_class: small` here as it's already defined in `executor: test-executor`

**Review by MazOneTwoOne — COMMENTED** [review]
Just some comments. I would approve, but I am the author of the PR.

File: Rakefile
Comment (willc-work) [reply Rakefile:1]: its not an empty rakefile, the changes to the file were modifying the permissions

File: .circleci/config.yml (hunk 1 above)
Comment (willc-work) [reply .circleci/config.yml:281]: yes looks like duplication so good to remove it

File: Rakefile
Comment (willc-work) [reply Rakefile:1]: but those modifications turned out to be unnecessary so I have reverted them

**Review by exonian — APPROVED** [review]
Looks good. Speeds things up well, and we can tune the number of nodes if the docker layer caching PR automatically speeds up the setup. If not, we can make optimisation its own ticket.

**Pull request merged**